pkgs/applications/editors/vscode/extensions/default.nix

```diff
@@ -3210,7 +3211,27 @@ let
 
       ms-vscode-remote.remote-ssh = callPackage ./ms-vscode-remote.remote-ssh { };
 
-      ms-vsliveshare.vsliveshare = callPackage ./ms-vsliveshare.vsliveshare { };
+      ms-vsliveshare.vsliveshare = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "vsliveshare";
+          publisher = "ms-vsliveshare";
+          version = "1.0.5918";
+          hash = "sha256-Tk0mKydUF8M7l7NC9wEA7t2rzJWy/mq4/HvIHI2/ldQ=";
+        };
+
+        postPatch = ''
+          substituteInPlace extension.js \
+            --replace-fail '"xsel"' '"${xsel}/bin/xsel"'
+        '';
+
+        meta = {
+          description = "Real-time collaborative development for VS Code";
+          homepage = "https://aka.ms/vsls-docs";
+          changelog = "https://marketplace.visualstudio.com/items/MS-vsliveshare.vsliveshare/changelog";
```

Comment:
Usually (or hopefully) alphabetical order is followed in attrs and lists. There are exceptions like: `meta.broken` which is always last to avoid git conflicts.
Considering alphabetical order, `changelog` would be before `description`.

Reply:
This isn't true; attributes appear in a logical order in derivations. For instance, typically `pname` and then `version` can be found, then `src`, patches, and so on; and `meta` can be found at the end. Likewise, the meta has its own ordering: `changelog` appears after `homepage` and before `license`; this can be seen [in the Nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/#var-meta-changelog).

Reply:
I did not see this message when I wrote that message. Would be good if you did not (prematurely) mark conversation as resolved and let the reviewer do that. Because it hides the message unless the reviewer clicks on it. In your mind, I understand your argument is reasonable to you, but the other person has to catch up. Consensus is reached not when you provide a sound argument, but when the other part has reached the same understanding. And then, marking resolved makes the most sense for the reviewer.

---

Yes, attributes are in a logical order in derivations following phases order. I did not refer to that. And I did say there are exceptions. Which was not the case. And you did correct me right. VSCode just happens to not be following that. As you noticed. What I was trying to say is that it is good to use alphabetical order whenever possible in attrs/lists.

Reply:
Hi, sorry for unintentionally closing these without you getting to read them. I was incorrectly assuming that you were getting notifications for them and had seen them. For instance, I get an e-mail for each individual comment you make, which immediately shows up on my phone. I was also under this impression because there was a review I had responded to (which had not yet been marked as resolved) which you just didn't react to at all. So it was not clear that this was important.

I like to mark things as resolved once I've handled them, since not doing so has the possibility of producing situations where a reviewer will come in and comment on something I've done, I'll handle it, but they'll disappear and not respond further, making other reviewers less likely to touch the PR or merge it. This is quite annoying, I'm sure you'll appreciate.